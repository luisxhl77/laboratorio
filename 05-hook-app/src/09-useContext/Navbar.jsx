import { Link, NavLink } from "react-router-dom"

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg bg-dark navbar-dark">
            <div className="container-fluid">

                <Link className="navbar-brand" to="/">useContext</Link>

                <div className="collapse navbar-collapse" id="navbarNav">

                    <ul className="navbar-nav">
                    
                    <NavLink to ="/" className={(args) => {
                        return 'nav-link';
                    } }>
                        Home
                    </NavLink>

                    <NavLink to ="/login" className={(args) => {
                        return 'nav-link';
                    } }>
                        Login
                    </NavLink>

                    <NavLink to ="/about" className={(args) => {
                        return 'nav-link';
                    } }>
                        About
                    </NavLink>

                    </ul>

                </div>
            </div>
        </nav>
    )
}
