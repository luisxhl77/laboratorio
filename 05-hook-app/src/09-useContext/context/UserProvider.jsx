import { useState } from "react"
import { Usercontext } from "./userContext"

const user = {
    id: 123,
    name: 'fernando herrera',
    email: 'herrera@gmail.com',
}

export const UserProvider = ({ children }) => {

    const [user, setUser] = useState();

  return (
        <Usercontext.Provider value={{user, setUser}}>
            {children}
        </Usercontext.Provider>
    )
}
