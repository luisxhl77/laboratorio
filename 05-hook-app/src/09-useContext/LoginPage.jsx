import { useContext } from "react"
import { Usercontext } from "./context/userContext";

export const LoginPage = () => {

    const { user, setUser} = useContext( Usercontext );
    return (
        <>
            <h1>LoginPage</h1>
            <hr />
            <pre aria-label="pre">
                {JSON.stringify(user, null, 3) }
            </pre>
            <button aria-label="btnSetUser" className="btn btn-primary" onClick={ () => setUser({id: 123, name: 'juan', email: 'asjda@gmail.com'})}>

                establecer usuario
            </button>
        </>
    )
}