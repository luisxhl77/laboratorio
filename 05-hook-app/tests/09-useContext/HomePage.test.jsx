import { render, screen } from "@testing-library/react"
import { Usercontext } from "../../src/09-useContext/context/userContext";
import { HomePage } from "../../src/09-useContext/HomePage"

describe('Pruebas de <HomePage/>', () => {

    const user = {
        id: 1,
        name: 'fernando'
    }

    test('debe de hacer match con el snapshot', () => {

        const { container } = render( 
            <Usercontext.Provider value={{ user:null  }}>
                <HomePage/> 
            </Usercontext.Provider>
        );
        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar el componente sin el usuario', () => {

        render( 
            <Usercontext.Provider value={{ user: null  }}>
                <HomePage/> 
            </Usercontext.Provider>
        );
        const preTag = screen.getByLabelText('pre');
        
        expect( preTag.innerHTML ).toBe('null');
    })

    test('debe de mostrar el componente con el usuario', () => {

        render( 
            <Usercontext.Provider value={{ user }}>
                <HomePage/> 
            </Usercontext.Provider>
        );
        const preTag = screen.getByLabelText('pre');
        
        expect( preTag.innerHTML ).toContain(user?.name);
        expect( preTag.innerHTML ).toContain(user?.id.toString());
    })
})