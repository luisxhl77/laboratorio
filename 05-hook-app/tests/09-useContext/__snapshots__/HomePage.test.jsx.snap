// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <HomePage/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    HomePage 
    <small />
  </h1>
  <hr />
  <pre
    aria-label="pre"
  >
    null
  </pre>
</div>
`;
